src/backend/XmlHttpRequest.ts

```diff
@@ -167,7 +192,7 @@ export default class XmlHttpRequest extends BaseFileSystem implements FileSystem
   }
 
   public supportsSynch(): boolean {
-    return true;
+    return (typeof(XMLHttpRequest) !== "undefined" && XMLHttpRequest !== null);
```

Comment:
What do you think about adding this and the `fetch` check into `xhr.ts` and `fetch.ts` as exported `const` booleans?

e.g.

```typescript
export const xhrIsAvailable = typeof(XMLHttpRequest) !== "undefined" && XMLHttpRequest !== null;
```

(Like the other comment, I can do this post-merge.)